app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/room/RoomDatasetWithCategories.kt

```diff
@@ -0,0 +1,15 @@
+package com.github.HumanLearning2021.HumanLearningApp.room
+
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.Relation
+
+data class RoomDatasetWithCategories(
+    @Embedded val dataset: RoomDataset,
```

Comment:
why is this one embedded?

Reply:
RoomDatasetWithCategories is the equivalent to Dataset. The naming might not be the best. I will rename it to make it clearer why.